pkg/proxy/iptables/proxier.go

```diff
@@ -818,7 +818,7 @@ func (proxier *Proxier) syncProxyRules() {
 	defer func() {
 		if !success {
 			proxier.logger.Info("Sync failed", "retryingTime", proxier.syncPeriod)
-			proxier.syncRunner.RetryAfter(proxier.syncPeriod)
+			proxier.syncRunner.Run()
```

Comment:
I think you're confusing `syncPeriod` and `minSyncPeriod`.

The goal is to try again but specifically _not_ to try again as soon as possible, since this failure is supposed to be a "can't happen" thing, so we assume if we try again immediately we might hit the same error again. Especially, in the iptables case, if someone else is holding the xtables lock for a really long time and we time out trying to acquire it, we want to wait and give them time to finish what they were doing.

Reply:
Yes. `RetryAfter(x)` effectively means "if `x` elapses without a sync happening, then sync". (The same way that the construct-time `maxInterval` means "if `maxInterval` elapses without a sync happening, then sync".)

`RetryAfter` was added a while after the rest of BFR, so it's not as well-integrated with the code as it might have been. (It's also tricky because we wanted to make sure that it works both from inside and from outside a sync, which made the locking tricky.)